src/json.rs

```diff
@@ -0,0 +1,426 @@
+// Vendored from `smoljson` bef592c5da1c3fe38b2462a8d231b0e0c8a86f80, with explicit permission
+// from the author (`thomcc`). Minimized for cc/simplicity. Modifications and additions made to fit cc's needs.
+
+use std::borrow::Cow;
+
+#[derive(PartialEq, Debug, Clone)]
+pub(crate) enum Token<'s> {
+    Null,
+    Bool(bool),
+    NumU(u64),
+    NumI(i64),
+    NumF(f64),
+    StrBorrow(&'s str),
+    StrOwn(Box<str>),
+    Colon,
+    Comma,
+    ObjectBegin,
+    ObjectEnd,
+    ArrayBegin,
+    ArrayEnd,
+}
+
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub struct Error(());
+
+impl core::fmt::Display for Error {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        f.write_str("JSON parse error")
+    }
+}
+
+pub type Result<T, E = Error> = core::result::Result<T, E>;
+
+pub(crate) struct Reader<'a> {
+    input: &'a str,
+    bytes: &'a [u8],
```

Comment:
Do we need bytes here, when we know that it contains valid utf-8?

Reply:
Aha I see sometimes bytes does come handy, because it support indexing using usize